From ad307d7670b0fce95cf7744ef7f997eae183a5f1 Mon Sep 17 00:00:00 2001
From: Martin Sperl <kernel@martin.sperl.org>
Date: Tue, 17 May 2016 08:26:32 +0000
Subject: [PATCH 41/65] ASoC: hw-params-rules: fix return value for matcher

fix the expected return value for the matcher
(!= 0 if it matches)

Signed-off-by: Martin Sperl <kernel@martin.sperl.org>
---
 sound/soc/generic/hw-params-rules.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sound/soc/generic/hw-params-rules.c b/sound/soc/generic/hw-params-rules.c
index 90881ef..2115c90 100644
--- a/sound/soc/generic/hw-params-rules.c
+++ b/sound/soc/generic/hw-params-rules.c
@@ -206,6 +206,7 @@ static int asoc_generic_hw_params_handle_rule(
 	list_for_each_entry(am, &rule->matches, list) {
 		dev_dbg(dev, "\tRunning match %pf(%pK)\n",
 			am->method, am->data);
+		/* match method return 0 on match, 1 otherwise */
 		ret = am->method(substream, params, am->data);
 		if (!ret)
 			return 1;
@@ -215,10 +216,12 @@ static int asoc_generic_hw_params_handle_rule(
 	list_for_each_entry(am, &rule->actions, list) {
 		dev_dbg(dev, "\tRunning action %pf(%pK)\n",
 			am->method, am->data);
+		/* action method returns 0 on success */
 		ret = am->method(substream, params, am->data);
 		if (ret)
 			return ret;
 	}
+
 	return 0;
 }
 
-- 
2.7.4

